spec/unit/views/components/paginated_collection_spec.rb

```diff
@@ -91,6 +91,32 @@ def paginated_collection(*args)
       end
     end
 
+    context "when specifying download_url_options" do
+      let(:view) do
+        view = mock_action_view
+        allow(view.request).to receive(:query_parameters).and_return details: '1'
+        allow(view.request).to receive(:path_parameters).and_return  controller: 'admin/users', action: 'show', id: 1
+        view
+      end
+
+      let(:collection) do
+        posts = 10.times.map{ Post.new }
+        Kaminari.paginate_array(posts).page(1).per(5)
+      end
+
+      let(:pagination) { paginated_collection(collection,
+                                              download_url_options: {
+                                                controller: 'admin/posts',
+                                                action: 'index',
+                                                params: { user_id: 1 }
+                                              },
+                                              download_links: [:csv]) }
+
+      it "should generate download link URLs using download_url_options" do
+        expect(pagination.find_by_tag('a').last.attr(:href)).to eq('/admin/posts.csv?user_id=1')
```

Comment:
Line is too long. [96/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.